### ISSUE TITLE
[Bug]: Unable to close Filter popup without choosing a filter selection

### ISSUE THREAD

**Issue by sharat87** [appsmithorg/appsmith#8476]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

On a table widget, clicking the Filter button at the top opens a movable Filter popup. But then this popup is impossible to close, unless a selection for filtering is made.

https://user-images.githubusercontent.com/120119/136984927-9efc9a04-5502-4d2d-b654-4049a0c5d463.mov

### Steps To Reproduce

```markdown
Just open Filter box for a Table widget and try to close it.
```

### Environment

Production

### Version

Cloud

**Comment by tmdoit** [comment]
Yes, it's poor experience.

**Comment by somangshu** [comment]
This is fixed in #8734
